modules/utils/docker/bridge_settings.sh

```diff
@@ -50,20 +50,111 @@ if [ -f "$COMMS_PCB_VERSION_FILE" ]; then
 fi
 echo $eth_port
 
-# Set mtu back to 1500 to support e2e connectivity
-# TODO: Investigate this as it should still work with 1460
-ifconfig bat0 mtu 1500
-ifconfig br-lan mtu 1500
-
-brctl addif br-lan bat0 $eth_port
-echo $br_lan_ip
-ifconfig br-lan $br_lan_ip netmask "255.255.255.0"
-ifconfig br-lan up
-echo
-ifconfig br-lan
+create_dhcpd_config()
+{
+  SUBNET="$1"
+
+  cat > /etc/dhcp/dhcpd.conf <<- EOF
+  default-lease-time 600;
+  max-lease-time 7200;
+  ddns-update-style none;
+  authoritative;
+
+  subnet $SUBNET.0 netmask 255.255.255.0 {
+    range $SUBNET.100 $SUBNET.199;
+    option routers $SUBNET.1;
+  }
+EOF
+  cp /dev/null /var/lib/dhcp/dhcpd.leases
+}
+
+create_olsrd_config()
+{
+  wifidev="$1"
+  SUBNET="$2"
+  IPV6_PREFIX="$3"
+  cat > /etc/olsrd/olsrd.conf <<- EOF
+  LinkQualityFishEye   0
+
+  Interface "$wifidev"
+  {
+  }
+
+  IpVersion               4
+  LinkQualityFishEye      0
+  LinkQualityAlgorithm "etx_ffeth_nl80211"
+
+  LoadPlugin "/usr/lib/olsrd_arprefresh.so.0.1"  
+  {
+  }
+
+  Hna4
+  {
+    $SUBNET.0 255.255.255.0
+  }
+
+  Hna6
+  {
+    $IPV6_PREFIX:0 64
+  }
+EOF
+}
+
+create_radvd_config()
+{
+  IPV6_PREFIX="$1"
+  cat > /etc/radvd.conf <<- EOF
+  interface br-lan
+  {
+    AdvSendAdvert on;
+    MinRtrAdvInterval 3;
+    MaxRtrAdvInterval 10;
+    prefix $IPV6_PREFIX:0/64 {
+    };
+  };
+EOF
+}
+
+
+
+# Bridge ethernet and Mesh for Batman-avd and routing for QoS-OLSR
+if [ "$algo" = "olsr" ]; then
+  brctl addif br-lan "$eth_port"
+  IPV6_PREFIX=$( echo fd`dd if=/dev/urandom bs=7 count=1 status=none | xxd -p` | sed 's/\(....\)/\1:/g' )
+  ip addr add $SUBNET.1/24 dev br-lan && ip addr add $IPV6_PREFIX:1/64 dev br-lan
+  create_olsrd_config "wlp1s0" "$SUBNET" "$IPV6_PREFIX"
+  create_dhcpd_config "$SUBNET"
+  dhcpd -f br-lan
+  create_radvd_config "$IPV6_PREFIX"
+  br_lan_ip="$SUBNET."$((16#$ip_random))	
+  wlp1s0_ip="192.168.11."$((16#$ip_random))
+  ifconfig br-lan up
+  ifconfig wlp1s0 "$wlp1s0_ip" 
+  iptables -A FORWARD --in-interface $mesh_if -j ACCEPT
+  iptables -A FORWARD -i br-lan -o $mesh_if -j ACCEPT 
+  iptables -A FORWARD -o br-lan -i $mesh_if -j ACCEPT
+  killall olsrd 2>/dev/null	
+  (qos-olsrd -i wlp1s0 -d 0)&
+  ifconfig "$mesh_if" mtu 1500
+else
```

Comment:
the "else" block here should try and just use the code that is already there so there are fewer changes. In this case the lines are actually the same (just in a different order), but in future try to change as little as possible to make things easier to review.

Reply:
> 1. Is line 136 meant to kill the original OLSR daemon? If so, why do we have both installed?
> 2. Is line 136 (also) meant to kill the QoS-OLSR daemon? If so, it needs to be killall qos-olsrd or pkill olsrd :)
> 3. Is it ok that we are killing absolutely all instances instead of a specific PID?
> 4. According to olsrd manpage -d 0 makes it run in the background so line 137 should need neither () nor &
I believe this is just a hangover from before the daemon name change. @francislin12 please fix.